brave/brave-core
# [WIP] Migrate brave-core changes from Android Ads

Fixes https://github.com/brave/brave-browser/issues/5160

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

- Confirm ad notifications are shown
- Switch ads off and then back on and confirm ad notifications are still shown
- Confirm ads are cashed-in
- Confirm clicking an ad that it opens in a new tab
- Confirm `default/ads_service` state is removed when ads are switched off
- Confirm ad notifications are closed when switching ads off
- Confirm ads are not shown if the network is disconnected

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
